ipld/resolver/src/vote_record.rs

```diff
@@ -38,24 +39,21 @@ impl<'de> Deserialize<'de> for ValidatorKey {
 }
 
 /// Vote by a validator about the validity/availability/finality
-/// of a CID in a given subnet.
+/// of something in a given subnet.
 #[derive(Serialize, Deserialize, Debug, Clone, Eq, PartialEq)]
-pub struct VoteRecord {
+pub struct VoteRecord<C> {
     /// Public key of the validator.
     pub public_key: ValidatorKey,
-    /// The subnet in which the vote is valid, to prevent a vote on the same CID
+    /// The subnet in which the vote is valid, to prevent a vote on the same subject
     /// in one subnet being replayed by an attacker on a different subnet.
     pub subnet_id: SubnetID,
-    /// The CID of the content the vote is about.
-    pub cid: Cid,
-    /// The claim of the vote, in case there can be votes about multiple facets
-    /// regarding the CID.
-    pub claim: String,
```

Comment:
@aakoshh could you elaborate on these removals?

Reply:
Sure, when I originally added this feature it the story was that we want to vote on the availability of some piece of data, represented by a CID. 

For example the subnet signals that they want to submit a checkpoint by sending a CID with a quorum to the parent. Then, the nodes on the parent have to resolve the CID of the checkpoint to its contents via the Resolver. But a validator faces exactly the same dilemma we have in our current story: when should they propose it in a parent block for execution? If they propose it too early and the majority hasn't yet resolved the CID, they will not vote for the block, risking liveness. 

The escape hatch was this voting: let's gossip that _"I have the CID, safe to propose"_, and only propose once we have a quorum to support it. I added a `claim` just in case we wish to differentiate what we are saying about it, but it was just an idea. 

This PR replaces the CID with a custom content, in this case a block hash (which is a CID in case the parent is Lotus) and a block height. I thought if we make the content flexible like that, we might as well remove the `claim` and use an `enum` for all the various statements we can vote on, such as `HaveCid | ParentFinality | ...`.